### ISSUE TITLE
[BUG] Bloom inference hang up with ZeRO3 and multiple gpus

### ISSUE THREAD

**Issue by kisseternity** [microsoft/DeepSpeed#3550]
**Describe the bug**
Hello, when I was training BLOOM in DeepSpeed Chat in step3, I found that inference would always hang up with ZeRO3 and multiple gpus. ZeRO2 and single gpu are okay to go. 
As I search this probloem in history issues, I've found two issues similar to my situation:
https://github.com/microsoft/DeepSpeed/issues/860
https://github.com/huggingface/transformers/issues/16688

As @[stas00](https://github.com/stas00) have resolved these issues, could you please have a look at my issue? Thanks a lot.
**To Reproduce**
Using DeepSpeed Chat, set the model to bloom, run with ZeRO3 and multiple gpus. 
Besides, setting the generate function like
         generate_params = {
            'do_sample': True,
            'top_k':15,
            'top_p': 0.8,
            'max_new_tokens': self.max_answer_seq_len,
            'eos_token_id': self.tokenizer.eos_token_id,
            'no_repeat_ngram_size': 0,
            'repetition_penalty': 1.1,
            'temperature': 0.7,
            'num_return_sequences': 1,
            'output_scores': True,
            'return_dict_in_generate': True,
        }
         outputs = self.actor_model.module.generate(
                prompts,
                attention_mask=mask,
                **generate_params
            )
**Expected behavior**
The training step hang in generate func.

**ds_report output**
I use the original set in DeepSpeed Chat. I think only ZeRO3 is the key config for this issue.

**System info (please complete the following information):**
 - OS: Ubuntu 18.04
 - GPU count and types: 8 * A100 80GB
 - Interconnects (if applicable) Just one machine
 - Python version 3.9.0

**Launcher context**
Launching with mpirun, not deepspeed launcher.

**Additional context**
As things go all right when using single GPU, or without ZeRO3,  or using the original generation func(which will produce all the same token length for all GPUs), I think this issue is really similar to https://github.com/microsoft/DeepSpeed/issues/860.
